src/CSET/__init__.py

```diff
@@ -138,68 +171,43 @@ def main():
     )
     parser_recipe_id.set_defaults(func=_recipe_id_command)
 
-    cli_args = sys.argv[1:] + shlex.split(os.getenv("CSET_ADDOPTS", ""))
-    args, unparsed_args = parser.parse_known_args(cli_args)
-
-    # Setup logging.
-    logging.captureWarnings(True)
-    loglevel = calculate_loglevel(args)
-    logger = logging.getLogger()
-    logger.setLevel(min(loglevel, logging.INFO))
-    stderr_log = logging.StreamHandler()
-    stderr_log.addFilter(lambda record: record.levelno >= loglevel)
-    stderr_log.setFormatter(logging.Formatter("%(asctime)s %(levelname)s %(message)s"))
-    logger.addHandler(stderr_log)
-
-    # Down here so runs after logging is setup.
-    logging.debug("CLI Arguments: %s", cli_args)
-
-    if args.subparser is None:
-        print("Please choose a command.", file=sys.stderr)
-        parser.print_usage()
-        sys.exit(127)
-
-    try:
-        # Execute the specified subcommand.
-        args.func(args, unparsed_args)
-    except ArgumentError as err:
-        # Error message for when needed template variables are missing.
-        print(err, file=sys.stderr)
-        parser.print_usage()
-        sys.exit(127)
-    except Exception as err:
-        # Provide slightly nicer error messages for unhandled exceptions.
-        print(err, file=sys.stderr)
-        # Display the time and full traceback when debug logging.
-        logging.debug("An unhandled exception occurred.")
-        if logging.root.isEnabledFor(logging.DEBUG):
-            raise
-        sys.exit(1)
+    return parser
 
 
-def calculate_loglevel(args) -> int:
-    """Calculate the logging level to apply.
+def setup_logging(verbosity: int):
+    """Configure logging level, format and output stream.
 
     Level is based on verbose argument and the LOGLEVEL environment variable.
     """
+    logging.captureWarnings(True)
+
+    # Calculate logging level.
     try:
         name_to_level = logging.getLevelNamesMapping()
     except AttributeError:
         # logging.getLevelNamesMapping() is python 3.11 or newer. Using
         # implementation detail for older versions.
         name_to_level = logging._nameToLevel
     # Level from CLI flags.
-    if args.verbose >= 2:
-        loglevel = logging.DEBUG
-    elif args.verbose == 1:
-        loglevel = logging.INFO
+    if verbosity >= 2:
+        cli_loglevel = logging.DEBUG
+    elif verbosity == 1:
+        cli_loglevel = logging.INFO
     else:
-        loglevel = logging.WARNING
```

Comment:
Just thought before I forget, do we import iris Future? Otherwise throws up a lot of issues. We also might get some deprecation errors thrown up by iris, how do we handle these?

Reply:
Yep, we set the following:
https://github.com/MetOffice/CSET/blob/67ad3c4ae22e6edd9b78f51d6c5a397627ab60e8/src/CSET/operators/__init__.py#L60-L63